orangecanvas/application/canvasmain.py

```diff
@@ -1179,10 +1248,87 @@ def new_scheme_from(self, filename):
                      "of the widgets/links in the "
                      "workflow."
                 ),
-                details="\n".join(map(repr, errors))
+                details=details,
+                parent=self,
             )
         return new_scheme
 
+    def check_requires(self, fileobj: IO) -> bool:
+        requires = scheme_requires(fileobj, self.widget_registry)
+        requires = [req for req in requires
+                    if not addons.is_requirement_available(req)]
+        if requires:
+            details_ = [
+                "<h4>Required packages:</h4><ul>",
+                *["<li>{}</li>".format(escape(r)) for r in requires],
+                "</ul>"
+            ]
+            details = "".join(details_)
+            mb = QMessageBox(
+                parent=self,
+                objectName="install-requirements-message-box",
+                icon=QMessageBox.Question,
+                windowTitle="Install Additional Packages",
+                text="This workflow requires additional packages to run. "
+                     "<br/>" + details + "<br/>"
+                     "Would you like to install them now?",
+                standardButtons=QMessageBox.Ok | QMessageBox.Abort |
+                                QMessageBox.Ignore,
+                informativeText=(
+                    "After installation you will have to restart the "
+                    "application and reopen the workflow"),
+            )
+            mb.setDefaultButton(QMessageBox.Ok)
+            bignore = mb.button(QMessageBox.Ignore)
+            bignore.setText("Load Partial")
```

Comment:
I was a bit confused by "Load Partial" button. The tooltip explains it nicely, but I was thinking whether there's a more informative label for it. Perhaps there isn't.